pytorch/training/docker/2.0/py3/cu121/Dockerfile.gpu

```diff
@@ -0,0 +1,290 @@
+ARG PYTHON=python3
+ARG PYTHON_VERSION=3.10.8
+ARG PYTHON_SHORT_VERSION=3.10
+ARG MAMBA_VERSION=23.1.0-1
+ARG PYTORCH_VERSION_EC2=2.0.1
+ARG CONDA_CHANNEL=https://aws-ml-conda-ec2.s3.us-west-2.amazonaws.com
+
+FROM nvidia/cuda:12.1.0-base-ubuntu20.04 AS base_image
+
+ENV DEBIAN_FRONTEND=noninteractive \
+    LD_LIBRARY_PATH="${LD_LIBRARY_PATH}:/usr/local/lib"
+
+RUN apt-get update \
+ && apt-get upgrade -y \
+ && apt-get autoremove -y \
+ && apt-get clean \
+ && rm -rf /var/lib/apt/lists/*
+
+FROM base_image AS common
+
+LABEL maintainer="Amazon AI"
+LABEL dlc_major_version="2"
+
+ARG PYTHON
+ARG PYTHON_VERSION
+ARG PYTHON_SHORT_VERSION
+ARG MAMBA_VERSION
+ARG CONDA_CHANNEL
+ARG EFA_PATH=/opt/amazon/efa
+
+# This arg required to stop docker build waiting for region configuration while installing tz data from ubuntu 20
+ARG DEBIAN_FRONTEND=noninteractive
+
+# Python won’t try to write .pyc or .pyo files on the import of source modules
+# Force stdin, stdout and stderr to be totally unbuffered. Good for logging
+ENV CUDA_HOME=/opt/conda
+ENV PYTHONDONTWRITEBYTECODE=1
+ENV PYTHONUNBUFFERED=1
+ENV LD_LIBRARY_PATH="/usr/local/lib:${LD_LIBRARY_PATH}"
+ENV LD_LIBRARY_PATH="/opt/conda/lib:${LD_LIBRARY_PATH}"
+ENV PYTHONIOENCODING=UTF-8
+ENV LANG=C.UTF-8
+ENV LC_ALL=C.UTF-8
+ENV PATH /opt/conda/bin:$PATH
+# older archs are not supported on CUDA12.1 like 3.7 for P2 instance
+# 5.2 is G3 EC2 instance, 7.5 is G4*, 7.0 is p3*, 8.0 is P4*, 8.6 is G5* and 9.0 is P5*
+ENV TORCH_CUDA_ARCH_LIST="5.2;7.0+PTX;7.5;8.0;8.6;9.0"
+ENV TORCH_NVCC_FLAGS="-Xfatbin -compress-all"
+ENV NCCL_VERSION=2.18.3
+ENV EFA_VERSION=1.24.1
+ENV GDRCOPY_VERSION=2.3.1
+
+ENV CMAKE_PREFIX_PATH="$(dirname $(which conda))/../"
+ENV OPEN_MPI_PATH=/opt/amazon/openmpi
+
+ENV DGLBACKEND=pytorch
+ENV MANUAL_BUILD=0
+ENV RDMAV_FORK_SAFE=1
+ENV DLC_CONTAINER_TYPE=training
+
+RUN apt-get update \
+ # TODO: Remove systemd upgrade once it is updated in base image
+ && apt-get -y upgrade --only-upgrade systemd \
+ && apt-get install -y --allow-change-held-packages --no-install-recommends \
+    build-essential \
+    ca-certificates \
+    cmake \
+    curl \
+    emacs \
+    git \
+    hwloc \
```

Comment:
do we still need this hwloc if it comes with conda install aws pytorch now

Reply:
nope, will remove